src/data-access/repositories/groupRepository.ts

```diff
@@ -0,0 +1,79 @@
+import { DbError } from '../../core/errors/dbError.js';
+import { BaseGroup, Group } from '../../types/group.js';
+import { UserModel } from '../models/userModel.js';
+import { InitializeSequelize } from '../../database/postgreSQL/initializeSequelize.js';
+import { Model, Transaction } from 'sequelize';
+import { BaseRepository } from './baseRepository.js';
+
+
+export class GroupRepository extends BaseRepository {
+    constructor(groupModel: any, groupDataMapper: any) {
+        super(groupModel, groupDataMapper);
+        this.model = groupModel;
+        this.dataMapper = groupDataMapper;
+    }
+
+    async get(id: string): Promise<Group | undefined> {
+        const groupFromDB = await super.getEntityById(id);
+        return groupFromDB ? this.dataMapper.toDomain(groupFromDB.toJSON()) : undefined;
+    }
+
+    async getAll(): Promise<Array<Group | undefined>> {
+        let groupsFromDB;
+        try {
+            groupsFromDB = await this.model.findAll({
+                include: [
+                    {
+                        model: UserModel,
+                        as: 'users',
+                        attributes: ['id', 'login', 'password', 'age'],
+                        through: {
+                            attributes: []
+                        }
+                    }
+                ]
+            });
+        } catch (e) {
+            throw new DbError('Error retrieving group');
+        }
+        return groupsFromDB.map((group: Model) => {
+            return this.dataMapper.toDomain(group.toJSON());
+        });
+    }
+    async create(group: Group): Promise<Group> {
+        const groupToCreate = this.dataMapper.toDalEntity(group);
+        const createdGroup = await super.createEntity(groupToCreate);
+        return this.dataMapper.toDomain(createdGroup.toJSON());
+    }
+    async update(groupUpdates: BaseGroup, id: string): Promise<Group> {
+        const groupToUpdate = this.dataMapper.toDalEntity(groupUpdates);
+        const updatedGroup = await super.updateEntityById(groupToUpdate, id);
+        return this.dataMapper.toDomain(updatedGroup.toJSON());
+    }
+    async delete(id: string, groupToDelete: Group): Promise<Group> {
```

Comment:
the purpose of this method is just to delete entity from database. So, give that method chance to do that, why you use like proxy-parameter `groupTODelete` here?

```
async delete(id: string): Promise<IDeleteEntityData> {
   return super.deleteEntity(id);
}
```
(check what delete method in `sequelize` returns and map it to `IDeleteEntityData`)

What you did is very similar to custom case, i.e. to have a method which would delete and return some data. Doesn't it look like a business case? I think yes, it does! 
So, definitely such logic must be placed in service layer insted!